synapse/push/mailer.py

```diff
@@ -443,10 +443,15 @@ async def get_message_vars(
         if event.type != EventTypes.Message and event.type != EventTypes.Encrypted:
             return None
 
-        sender_state_event_id = room_state_ids[("m.room.member", event.sender)]
-        sender_state_event = await self.store.get_event(sender_state_event_id)
-        sender_name = name_from_member_event(sender_state_event)
-        sender_avatar_url = sender_state_event.content.get("avatar_url")
+        sender_state_event_id = room_state_ids.get(("m.room.member", event.sender))
+        if sender_state_event_id:
+            sender_state_event = await self.store.get_event(sender_state_event_id)
+            sender_name = name_from_member_event(sender_state_event)
+            sender_avatar_url = sender_state_event.content.get("avatar_url")
+        else:
+            # The sender is no longer in the room for some reason.
+            sender_name = event.sender
+            sender_avatar_url = None
```

Comment:
I wonder if we should use the state at the time of the event? Using the MXID is quite ugly so we should probably avoid it where possible.

I think we can just do:

```python
state = await self.state_store.get_state_for_event(StateFilter.from_types((("m.room.member", event.sender),))
````

Falling back to the sender MXID if that is also empty?

And then ditto below

Reply:
I don't think this works though since (in this case) you don't have an event ID to pass to `get_state_for_event`. Looking through the state store I don't see anything that lets you not pass in an event ID.

Reply:
Oh, I unhelpfully forgot to put the event ID in the example. I mean using `event.event_id` to get the state at the time of the event that we're pushing? Or is `event` something else?

Reply:
That worked! 🎉 

I'm having a bit of trouble figure out how to do this below though since we seem to lose the "event" at some point. (This code is quite confusing...)